commands/pack.js

```diff
@@ -265,7 +265,13 @@ function api(opts = {}) {
 	);
 
 	// Set any entry path override
-	if (opts.entry) helper.replaceMain(config, path.resolve(opts.entry));
+	if (opts.entry) {
+		try {
+			helper.replaceMain(config, JSON.parse(opts.entry));
+		} catch (e) {
+			helper.replaceMain(config, path.resolve(opts.entry));
+		}
+	}
```

Comment:
It would be better to just pass `opts.entry` from CLI and process the value from dev-utils.
I confirmed that path.resolve throws the same result from both CLI and dev-utils.